Assets/Mirror/Editor/PreprocessorDefine.cs

```diff
@@ -1,31 +1,22 @@
 using System.Collections.Generic;
-using System.Linq;
 using UnityEditor;
 
 namespace Mirror
 {
     static class PreprocessorDefine
     {
-        /// <summary>
-        /// Symbols that will be added to the editor
-        /// </summary>
-        public static readonly string[] Symbols = new string[] {
-            "MIRROR",
-            "MIRROR_1726_OR_NEWER"
-        };
-
         /// <summary>
         /// Add define symbols as soon as Unity gets done compiling.
         /// </summary>
         [InitializeOnLoadMethod]
         static void AddDefineSymbols()
         {
-            string definesString = PlayerSettings.GetScriptingDefineSymbolsForGroup(EditorUserBuildSettings.selectedBuildTargetGroup);
-            List<string> allDefines = definesString.Split(';').ToList();
-            allDefines.AddRange(Symbols.Except(allDefines));
-            PlayerSettings.SetScriptingDefineSymbolsForGroup(
-                EditorUserBuildSettings.selectedBuildTargetGroup,
-                string.Join(";", allDefines.ToArray()));
+            HashSet<string> defines = new HashSet<string>(PlayerSettings.GetScriptingDefineSymbolsForGroup(EditorUserBuildSettings.selectedBuildTargetGroup).Split(';'))
+            {
+                "MIRROR",
+                "MIRROR_1726_OR_NEWER"
```

Comment:
why do we have 1726_or_newer btw?

Reply:
No idea, but iirc that was the first version with defines

Reply:
maybe we want to remove it and add "MIRROR_1.3_OR_NEWER"

Reply:
dots are illegal in defines

Reply:
But still, this doesn't have anything to do with my PR

Reply:
you are both right of course.

The only potential problem is that this changes the order of the defines.   If the user has a bunch of defines,  we are shuffling them here.

Reply:
Is that an issue?

Reply:
I would rather avoid it,  yes.   Here are my defines for cubica:

CROSS_PLATFORM_INPUT;MYSQL;SERVER;CLIENT;ENABLE_PLAYFABSERVER_API;ENABLE_PLAYFABADMIN_API;MIRROR;MIRROR_1726_OR_NEWER

if you shuffle them,  it will be very annoying for me to add/remove defines because they will be all in random places I am not used to.

Reply:
Also, iirc HashSet only changes the order on remove, and when i tested this rn, it hasn't changed the order

Reply:
HashSet hashes the data and stores them in the hash order,  if you have numbers such as 1,2,3,4,   then they will typically look sorted,   with strings the order will be for all practical purposes random

Reply:
Only adding to it keeps the initial order

Reply:
Just test this in your project and see if it keeps the order